platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -342,28 +330,125 @@ public NodeId getCreatorId() {
         return creatorId;
     }
 
+    /**
+     * Get the address book round of the event.
+     *
+     * @return the address book round of the event
+     */
+    public long getAddressBookRound() {
+        return addressBookRound;
+    }
+
+    /**
+     * Get the event descriptor for the self parent.
+     *
+     * @return the event descriptor for the self parent
+     */
+    public EventDescriptor getSelfParent() {
+        return selfParent;
+    }
+
+    /**
+     * Get the event descriptors for the other parents.
+     *
+     * @return the event descriptors for the other parents
+     */
+    public List<EventDescriptor> getOtherParents() {
+        return otherParents;
+    }
+
+    /**
+     * Get the self parent generation
+     *
+     * @return the self parent generation
+     */
     public long getSelfParentGen() {
-        return selfParentGen;
+        if (selfParent == null) {
+            return -1;
+        }
+        return selfParent.getGeneration();
     }
 
+    /**
+     * Get the maximum generation of the other parents.
+     *
+     * @return the maximum generation of the other parents
+     * @deprecated this method should be replaced since there can be multiple other parents.
+     */
     public long getOtherParentGen() {
-        return otherParentGen;
+        if (otherParents == null || otherParents.isEmpty()) {
+            return -1;
+        }
+        if (otherParents.size() == 1) {
+            return otherParents.get(0).getGeneration();
+        }
+        return otherParents.stream()
+                .max((a, b) -> Long.compare(a.getGeneration(), b.getGeneration()))
+                .get()
+                .getGeneration();
     }
 
+    /**
+     * Get the hash of the self parent.
+     *
+     * @return the hash of the self parent
+     */
     public Hash getSelfParentHash() {
-        return selfParentHash;
+        if (selfParent == null) {
+            return null;
+        }
+        return selfParent.getHash();
     }
 
+    /**
+     * Get the hash of the other parent with the maximum generation.
+     *
+     * @return the hash of the other parent with the maximum generation
+     * @deprecated
+     * */
     public Hash getOtherParentHash() {
```

Comment:
Once again, not sure we will keep this method when we start using multiple other parents. But not harmful for now.